code/_onclick/hud/screen_alarms.dm

```diff
@@ -402,3 +403,20 @@ so as to remain in compliance with the most up-to-date laws."
 	name_pick.namepick_message = namepick_message
 	name_pick.role = role
 	name_pick.allow_numbers = allow_numbers
+
+/obj/abstract/screen/alert/robot/reset_self
+	name = "Reset your module"
+	desc = "Click here to reset your module"
+	icon_state = "module_reset"
+	timeout = 60 SECONDS
+	emph = TRUE
+
+/obj/abstract/screen/alert/robot/reset_self/Click()
+	..()
+	var/mob/living/silicon/robot/R = usr
+	if(/obj/item/borg/upgrade/vtec in R.module.upgrades)
+		R.movement_speed_modifier -= SILICON_VTEC_SPEED_BONUS
+	qdel(R.module)
+	R.set_module_sprites(list("Default" = "robot"))
+	qdel(src)
```

Comment:
```suggestion
	R.install_upgrade(R, /obj/item/borg/upgrade/reset)
	qdel(src)
```

This will fail if their modules can't be reseted normally(and eject the magical reset board too, but you can make a children of the board that deletes itself or something), but i think it's better this way.